blogposts/structural-code-search-at-scale-from-your-browser.md

```diff
@@ -0,0 +1,305 @@
+  the default branch. We plan to open it up to all mirrored repositories and you
+  can [help make that happen faster](#whats-next-for-structural-search).
+
+- You can [set up Sourcegraph
+  locally](https://docs.sourcegraph.com/#quickstart) for your own code or any
+  other repository you'd like and get all of the structural search goodness.
+  Running locally will also likely be faster than using [sourcegraph.com](https://sourcegraph.com/search?q=&patternType=structural).
+
+- You might be running structural search for the first time on a repo! 😎 If
+your query times out, give the page a refresh because we're probably warming up
+the cache for you.
+
+- See our [usage
+  documentation](https://docs.sourcegraph.com/user/search/structural) for more
+  help and the [Comby FAQ](https://comby.dev/#faq) for more details and known
+  limitations of the matching engine.
+
+**A note on regular expressions: How is structural search different?*
+
+Structural search is not a replacement for regexp search. It's another tool in
+your toolkit that works well for matching blocks of code or expressions, and
+simplifies catching buggy syntactic patterns. If you only want to find a simple
+string or pattern, consider using Sourcegraph's literal or regexp
+[search](https://sourcegraph.com/search), because these queries are typically
+much faster! For a more detailed breakdown, see the short comparison at the [end
+of this post](#comparing-structural-search-to-more-traditional-text-search).
+
+## What's next for structural search?
+
+We have more features and improvements planned. _If you want to see any of these
+features arrive more quickly, please +1 the related issue tracker so that we can
+prioritize our engineering to deliver them sooner!_
+
+- **Structural search enabled for all mirrored repositories.** We want structural
+  search to be available for _your_ repository, and not just the really popular
+  ones. [+1 this feature on GitHub](https://github.com/sourcegraph/sourcegraph/issues/8164)
+- **Regular expression support in structural holes.** We want to add support for
+  regexp syntax inside holes for refine search, like `\d+` to match only
+  numerical digits. [+1 this feature on GitHub](https://github.com/sourcegraph/sourcegraph/issues/8165)
+- **A richer query language.** There are ways to refine structural search with [rules](https://comby.dev/#advanced-usage) allowing richer queries. To get a taste of rules, have a look at this [CactusCon talk](https://www.youtube.com/watch?v=yOZQsZs35FA) (subtitles recommended if audio is hard to follow). If you have a use case for this and want support sooner, [+1 this feature on GitHub](https://github.com/sourcegraph/sourcegraph/issues/8166).
+- **Make it faster.** Structural search is typically slower than our regexp
+  search because it does more work. If you find it valuable, we
+  want to make it faster. [+1 this feature on GitHub](https://github.com/sourcegraph/sourcegraph/issues/8167)
```

Comment:
These work items aren't planned any time soon. The main reason this is here is that I'd like to open up some way to gather a validation signal if people find certain features useful or missing.